ingredient/ingredient-postgresql/src/plugin.ts

```diff
@@ -1,28 +1,143 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
 import { BaseIngredient, IngredientManager,IIngredient,DeploymentContext } from "@azbake/core"
 import { ARMHelper } from "@azbake/arm-helper"
-import ARMTemplate from "./FlexibleServerArm.json" 
+import { PostgreSQLDBUtils } from "./functions"
+import { VnetData } from "./vnetData"
+import PublicAccessARMTemplate from "./PublicAccessArm.json" 
+import PrivateAccessARMTemplate from "./PrivateAccessArm.json"
+import { Subnet,VirtualNetwork } from "@azure/arm-network/esm/models"
 
 export class PostgreSQLDB extends BaseIngredient {
 
     constructor(name: string, ingredient: IIngredient, ctx: DeploymentContext) {
         super(name, ingredient, ctx);
         this._helper = new ARMHelper(this._ctx);
+        this._functions = new PostgreSQLDBUtils(this._ctx);
+        this._access = this._ingredient.properties.parameters.get("access")?._value.toLowerCase();
+        if (this._access == "public") {
+            this._armTemplate = PublicAccessARMTemplate
+        } else if (this._access == "private") {
+            this._armTemplate = PrivateAccessARMTemplate;
+        } else throw new Error("Parameter 'access' must be set to \"public\" or \"private\".");
     }
 
     _helper: ARMHelper;
+    _functions: PostgreSQLDBUtils; 
+    private _access: string;
+    private _armTemplate: any;
 
     public async Execute(): Promise<void> {
+        let params: any;
         try {
-            let util = IngredientManager.getIngredientFunction("coreutils", this._ctx)
-            this._logger.log('PostgreSQL Plugin Logging: ' + this._ingredient.properties.parameters)
+            params = await this._helper.BakeParamsToARMParamsAsync(this._name, this._ingredient.properties.parameters)
+            this.validateBakeParams(params);
+        } 
+        catch (error){
+            this._logger.error('Bake validation failed: ' + error)
+            throw error;
+        }
+
+        if (this._access == "private")
+        {
+            params.vnetData = await this.getVnetData(params);
 
-            let params = await this._helper.BakeParamsToARMParamsAsync(this._name, this._ingredient.properties.parameters)
+            // The Private ARM template includes a few Microsoft.Resources/deployments which should be uniquely named
+            const timestamp = new Date().toISOString().replace(/[^a-zA-Z0-9]/g, "");
+            params.virtualNetworkDeploymentName = {value: `virtualNetwork_${timestamp}`};
```

Comment:
don't create unique deployment names. You will fill up the deployments that are never deleted. Reuse a deployment name that is stable.

Reply:
Ah, okay. I was copying the pattern from the portal deployment template which named deployments with a timestamp so I assumed they weren't to be shared. I'll fix.